pkg/codegen/schema/bind.go

```diff
@@ -1359,6 +1369,19 @@ func (t *types) bindResourceDetails(path, token string, spec ResourceSpec, decl
 		return diags, fmt.Errorf("failed to bind properties for %v: %w", token, err)
 	}
 
+	// urn is a reserved property name for all resources
+	// id is a reserved property name for resources which are not components
+	// emit a warning if either of these are used
+	for _, property := range properties {
+		if property.Name == "urn" {
+			diags = diags.Append(warningf(path+"/properties/urn", "urn is a reserved property name"))
+		}
+
+		if !spec.IsComponent && property.Name == "id" {
+			diags = diags.Append(warningf(path+"/properties/id", "id is a reserved property name for resources"))
```

Comment:
```suggestion
			diags = diags.Append(warningf(path+"/properties/id", "id is a reserved property name for custom resources"))
```

Reply:
I don't know if it is clear enough for users to tell what _custom_ resources are. Aren't these "just resources" from their perspective? 🤔

Reply:
Yeh maybe, I whacked approve on this because I wasn't sure myself but thought worth tagging to see what anyone else thought

Reply:
An alternative would be to say "non-component resources" rather than "custom resources". Not sure if that's any more clear (although it does align with the `isComponent` property in the schema). I don't feel strongly between any of the three choices.

Reply:
To avoid any confusion, will keep it as is for the time being